tests/runner/src/main/java/org/mule/test/runner/utils/RunnerModuleUtils.java

```diff
@@ -94,16 +121,495 @@ public static void assureSdkApiInClassLoader(ClassLoader extensionClassLoader,
 
         Method method = findMethod(extensionClassLoader.getClass(), "addURL", URL.class);
 
+        // Until Java8, this will be the path taken.
         if (method != null) {
           method.setAccessible(true);
           method.invoke(extensionClassLoader, sdkApiUrl);
+          return;
         }
+
+
+        // For Java 9+ Use bytebuddy to add dynamically all classes from the Jar
+        addSdkApiClassesDynamically(extensionClassLoader, sdkApiUrl);
+
       } catch (Exception e) {
         throw new RuntimeException("Could not assure sdk-api in extension classloader", e);
       }
     }
   }
 
+  private static void addSdkApiClassesDynamically(ClassLoader extensionClassLoader, URL sdkApiUrl) throws IOException {
+    boolean classAdded = true;
+    Set<String> addedClasses = new HashSet<>();
+
+    // Some classes needs to be loaded in certain order and will fail to do so if tried to be loaded first.
+    // This code will try to load all the remaining classes and keep trying until none can be loaded.
+    while (classAdded) {
+      classAdded = false;
+      ZipInputStream zipInputStream = new ZipInputStream(new FileInputStream(sdkApiUrl.getFile()));
+      for (Map.Entry<TypeDescription, byte[]> classEntry : getSdkClassMap(zipInputStream, addedClasses).entrySet()) {
+
+        try {
+          Map<TypeDescription, byte[]> mapEntry = new HashMap<>();
+          mapEntry.put(classEntry.getKey(), classEntry.getValue());
+          ClassLoadingStrategy.Default.INJECTION.load(extensionClassLoader, mapEntry);
+          addedClasses.add(classEntry.getKey().getName());
+          classAdded = true;
+        } catch (Exception e) {
+          // Problem found adding class to class loader.
```

Comment:
Don't hide the exception. At the very least log it, but why wouldn't it be thrown?

Reply:
Some classes need to be added in certain order to the class loader, so in some cases this is expected to fail, and the class will be loaded on another try. I can log this in debug level if you think that is the correct approach.

Reply:
Yes. If that's the case, you should make a DEBUG log when that happens and you should also DEBUG log when a class is successfully loaded, so that you can actually verify if it worked on the next attempt. Any exception other than that should result in failure